mkosi/backend.py

```diff
@@ -948,3 +949,25 @@ def complete_step(cls, text: str, text2: Optional[str] = None) -> Iterator[List[
 
         if text2 is not None:
             cls.print_step(text2.format(*args))
+
+
+def chown_to_running_user(path: PathString) -> None:
+    uid = int(os.getenv("SUDO_UID") or os.getenv("PKEXEC_UID") or str(os.getuid()))
+    user = pwd.getpwuid(uid).pw_name
+    gid = pwd.getpwuid(uid).pw_gid
+
+    with MkosiPrinter.complete_step(
+        f"Changing ownership of output file {path} to user {user}…",
+        f"Changed ownership of {path}",
+    ):
+        os.chown(path, uid, gid)
+
+
+def mkdir_chown_current_user(path: Path, skip_chown: bool = False, mode: int = 0o777) -> None:
```

Comment:
The default 
```suggestion
def mkdir_chown_current_user(path: Path, skip_chown: bool = False, mode: int = 0o511) -> None:
```
The mode `0o777` is neither what was commonly used previously (`0o755`) nor the default of `Path.mkdir` (`0o511`). To not move to widely from pathlib I'd recommend that, so one doesn't have to think too much about semantics.

Reply:
[`pathlib.mkdir`'s default is `0o777`](https://docs.python.org/3.9/library/pathlib.html#pathlib.Path.mkdir)?

Reply:
My bad... `0o777` is 511 and I only had a quick look at the help output in my shell and didn't think twice.